flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/sink/TestReloadingTableSupplier.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.flink.sink;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.flink.TableLoader;
+import org.awaitility.Awaitility;
+import org.junit.Test;
```

Comment:
this should be a JUnit5 test

Reply:
The Flink tests are still using `org.junitTest` so I kept it consistent with those.

Reply:
for new test classes being added we actually want to use JUnit5, as otherwise this will just cause more work later on in the migration process.

You might need to add the below snippet to the build file (if it isn't already defined for the flink project)
```
test {
    useJUnitPlatform()
  }
```

Reply:
OK thanks, I made this change.